qa/rpc-tests/wallet_golden_5_6_0.py

```diff
@@ -32,6 +32,7 @@ def __init__(self):
     def setup_nodes(self):
         return start_nodes(self.num_nodes, self.options.tmpdir, extra_args=[[
             nuparams(NU5_BRANCH_ID, 201),
+            "-regtestwalletsetbestchaineveryblock",
```

Comment:
It would be helpful for the commit message of this first commit to clarify that this is fixing a bug in the golden test, where not all state was being saved (which then makes it clear to the reader why the cached golden state is changing).